src/main/java/org/skriptlang/skript/bukkit/spark/MsptStatistic.java

```diff
@@ -0,0 +1,121 @@
+package org.skriptlang.skript.bukkit.spark;
+
+import me.lucko.spark.api.statistic.misc.DoubleAverageInfo;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * An instance of a statistic to use with the {@link org.skriptlang.skript.bukkit.spark.expressions.ExprMSPT} expression.
+ * Can be an average, median, min, max, or specific percentile.
+ */
+public class MsptStatistic {
+
+	/**
+	 * The type of a MSPT statistic.
+	 */
+	private enum MsptStatisticType {
+		AVERAGE("(?:average|mean)", DoubleAverageInfo::mean),
+		MEDIAN("median", DoubleAverageInfo::median),
+		MAX("max(?:imum)?", DoubleAverageInfo::max),
+		MIN("min(?:imum)?", DoubleAverageInfo::min),
+		PERCENTILE("(\\d{1,2}(?:\\.\\d+)?)(?:st|nd|rd|th)\\s+percentile", DoubleAverageInfo::percentile);
+
+		private final Pattern pattern;
+		private @UnknownNullability Function<DoubleAverageInfo, Double> function;
+		private @UnknownNullability BiFunction<DoubleAverageInfo, Double, Double> percentileFunction;
+
+		MsptStatisticType(String pattern, Function<DoubleAverageInfo, Double> function) {
+			this.pattern = Pattern.compile(pattern);
+			this.function = function;
+		}
+
+		MsptStatisticType(String pattern, BiFunction<DoubleAverageInfo, Double, Double> percentileFunction) {
+			this.pattern = Pattern.compile(pattern);
+			this.percentileFunction = percentileFunction;
+		}
```

Comment:
this can be merged into one constructor

Reply:
What benefit would that give? Then we'd have to have nullable inputs and it'd be a bit messy.